packages/url/src/clean-for-slug.js

```diff
@@ -23,11 +23,9 @@ export function cleanForSlug( string ) {
 	if ( ! string ) {
 		return '';
 	}
-	return trim(
-		deburr( string )
-			.replace( /[\s\./]+/g, '-' )
-			.replace( /[^\p{L}\p{N}_-]+/gu, '' )
-			.toLowerCase(),
-		'-'
-	);
+	return removeAccents( string )
+		.replace( /[\s\./]+/g, '-' )
+		.replace( /[^\p{L}\p{N}_-]+/gu, '' )
+		.toLowerCase()
+		.replace( /(^-)|(-$)/g, '' );
```

Comment:
Is it about just a single trailing `-`, or all trailing `-`s?
```suggestion
		.replace( /(^-+)|(-+$)/g, '' );
```

Reply:
Good catch! 

If the string actually starts with a space, then is followed by a `-`, or ends with a `-` followed by a space, we'll end up with multiple `-` at the beginning or end. I'll add a test to cover that use case.

Reply:
Test added in https://github.com/WordPress/gutenberg/pull/41687/commits/f2f238a9e5c33313751e6d7dea99d047cb67ac2c. Thanks again!